src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp

```diff
@@ -485,12 +489,16 @@ real64 SurfaceGenerator::solverStep( real64 const & time_n,
       if( fluxApprox!=nullptr )
       {
         fluxApprox->addToFractureStencil( meshLevel, this->m_fractureRegionName, true );
-        FaceElementSubRegion & subRegion = fractureRegion.getSubRegion< FaceElementSubRegion >( 0 );
-        subRegion.m_recalculateFractureConnectorEdges.clear();
-        subRegion.m_newFaceElements.clear();
       }
     }
 
+    FaceElementSubRegion & subRegion = fractureRegion.getSubRegion< FaceElementSubRegion >( 0 );
+    subRegion.m_recalculateFractureConnectorEdges.clear();
+    subRegion.m_newFaceElements.clear();
```

Comment:
So now you're cleaning the `subRegion` each time, even if there's no `FluxApproximationBase`?
Should we check that there's a `FluxApproximationBase` before cleaning?

Reply:
I am not sure it actaully matters. The goal was actually to clean it after the loop over the FluxApproximationBase because when doing Poromechanics with Langrange multiplier contact there are 2 fluxApprox objects (2 tpfa stencils basically) and that info is the same for both so if it gets cleared once used by the first one the second one has an issue.